lua/flatten/guest.lua

```diff
@@ -20,20 +21,14 @@ local function send_files(host, files, stdin)
 		cwd = sanitize(cwd)
 	end
 
-	local call = string.format([[
-		return require('flatten.core').edit_files(
-			%s,   -- `args` passed into nested instance.
-			'%s', -- guest default socket.
-			'%s', -- guest global cwd.
-			%s,   -- stdin lines or {}.
-			%s    -- enable blocking
-		)]],
-		vim.inspect(files),
-		server,
-		cwd,
-		vim.inspect(stdin),
-		force_block and 'true' or 'false'
-	)
+	local call = string.format([[return require('flatten.core').edit_files(%s)]], vim.inspect({
+		files = files,
```

Comment:
> Do we need to send files here if we're also sending vim.v.argv?

I thought about it, but I kind of like the idea of having a list of just the files without having to categorize `v:argv` manually.

> May want to look into splitting vim.v.argv into files and cmds, etc. in guest.lua before they're sent to the host

The only thing that stopped me from doing that is that I don't know if there's any scenarios where we want to preserve the order of the commands and files.

It may not be an issue, though. I quickly ran a test by starting nvim with:

```shell
nvim file-a.txt -O file-b.txt -o file-c.txt
```

Expecting it to first create a vertical split, then a horizontal one. But it looks like neovim doesn't support that, it only cares about that last `-o`.

Reply:
That makes sense! Definitely worth doing more testing before making that change.